subsys/net/ip/Kconfig.tcp

```diff
@@ -231,7 +231,8 @@ config NET_TCP_ISN_RFC6528
 	depends on NET_TCP
 	select MBEDTLS
 	select MBEDTLS_MD
-	select MBEDTLS_MAC_MD5_ENABLED
+	select MBEDTLS_MAC_SHA256_ENABLED if !BUILD_WITH_TFM
```

Comment:
Tied into my previous comment. We *could maybe do a simple select of `PSA_WANT_ALG_SHA_256` without the depends statements and skip `select MBEDTLS_MAC_SHA256_ENABLED` altogether....